recipe/conda_build_config.yaml

```diff
@@ -4,5 +4,5 @@ cxx_compiler:      # [osx]
 - clang_bootstrap  # [osx]
 
 with_icu:
-- yes     # [not win]
```

Comment:
@isuruf can you comment on this here? I recall that you added this for a reason, right?

Reply:
It's fine to keep that part as-is, but then we need to not apply the `icu <0.0` run-constraint on windows. Here's a concrete [example](https://github.com/conda-forge/libpdal-feedstock/pull/17) that fails due the constraint.